x-pack/test/functional/apps/maps/import_geojson/file_indexing_panel.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import path from 'path';
+import uuid from 'uuid/v4';
+
+export default function ({ getService, getPageObjects }) {
+  const PageObjects = getPageObjects(['maps', 'common']);
+  const testSubjects = getService('testSubjects');
+  const log = getService('log');
+
+  const IMPORT_FILE_PREVIEW_NAME = 'Import File';
+  const FILE_LOAD_DIR = 'test_upload_files';
+  const DEFAULT_LOAD_POINT_FILE_NAME = 'point.json';
+
+  const indexPoint = async () => await loadFileAndIndex(DEFAULT_LOAD_POINT_FILE_NAME);
+
+  async function loadFileAndIndex(loadFileName) {
+    log.debug(`Uploading ${loadFileName} for indexing`);
+    await PageObjects.maps.uploadJsonFileForIndexing(
+      path.join(__dirname, FILE_LOAD_DIR, loadFileName)
+    );
+    await PageObjects.maps.waitForLayersToLoad();
+    await PageObjects.maps.doesLayerExist(IMPORT_FILE_PREVIEW_NAME);
+    await PageObjects.maps.hasFilePickerLoadedFile(loadFileName);
+
+    const indexName = uuid();
+    await PageObjects.maps.setIndexName(indexName);
+    await PageObjects.maps.clickImportFileButton();
+    return indexName;
+  }
+
+  describe('On index name & pattern operation complete', () => {
+    before(async () => {
+      await PageObjects.maps.openNewMap();
+    });
+
+    beforeEach(async () => {
+      await PageObjects.maps.clickAddLayer();
+      await PageObjects.maps.selectGeoJsonUploadSource();
+    });
+
+    afterEach(async () => {
+      await PageObjects.maps.cancelLayerAdd();
+      await PageObjects.maps.waitForLayerAddPanelClosed();
+    });
+
+    it('should not activate add layer button until indexing succeeds', async () => {
+      await indexPoint();
+
+      let layerAddReady = await PageObjects.maps.importFileButtonEnabled();
+      expect(layerAddReady).to.be(false);
+
+      layerAddReady = await PageObjects.maps.importLayerReadyForAdd();
+      expect(layerAddReady).to.be(true);
+    });
+
+    it('GeoJson file layer removed, ES search layer added', async () => {
```

Comment:
This test is a really implementation specific. Maybe rename to "Should load geojson file as ES document source layer". Then just verify the list list of layers equals the expected value.

The fact that the implementation adds 2 sources is a detail that does not need to be tested. The outcome of uploading geojson file results in a layer with ES results is what needs to be tested. You should also add an expect clause that uses inspector to verify the data is pulled from elasticsearch as expected

Reply:
I do like verifying inspector results so I've added that in. I'd prefer to leave in the details around verifying the 2 different sources. It wouldn't be hard to mess up the client Maps code and accidentally add both layers, not add the second, etc.